### PR TITLE
Fix orig files for conflicting generated

### PR DESCRIPTION
**Pull request by thinkerou**
Add *.orig in .gitignore file, and add orig file screening.
Issue: #7280

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -96,3 +96,6 @@ Pods/
 
 # Artifacts directory
 artifacts/
+
+# Git generated files for conflicting
+*.orig
```
Comment (nicolasnoble) [inline .gitignore]: You probably want **/*.orig

File: tools/run_tests/sanity/check_orig_mk.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+set -e
+
+if [ -f *.orig ] ; then
```
Comment (nicolasnoble) [inline tools/run_tests/sanity/check_orig_mk.sh]: I don't believe this test is necessary when we have the .gitignore properly in place. Or it needs to find files recursively.

File: .gitignore (hunk 1 above)
Comment (thinkerou) [inline .gitignore]: No, I want to ignore all files for terminating `orig`, so I think it should be `*.orig`, right?

File: tools/run_tests/sanity/check_orig_mk.sh (hunk 2 above)
Comment (thinkerou) [inline tools/run_tests/sanity/check_orig_mk.sh]: Yeth, so should I. This copy `check_cache_mk.sh`, I should delete the file and recommit?

File: .gitignore (hunk 1 above)
Comment (nicolasnoble) [inline .gitignore]: Github mangled my usage of stars. I meant `**/*.orig`

File: tools/run_tests/sanity/check_orig_mk.sh (hunk 2 above)
Comment (nicolasnoble) [inline tools/run_tests/sanity/check_orig_mk.sh]: Yeah, you can nuke it, I'm fine with that :)

Comment (thinkerou) [inline tools/run_tests/sanity/check_orig_mk.sh]: OK, thanks.

File: .gitignore (hunk 1 above)
Comment (thinkerou) [inline .gitignore]: But I test with `*.orig`, it can recursive ignore orig file.

**Pull request merged**